File: rust/word_count.rs
use std::io::{Read, Write};

fn main() {
    let mut input = Vec::new();
    std::io::stdin().read_to_end(&mut input).unwrap();

    let mut cnt: u32 = 0;
    let mut in_word = false;

    for &c in &input {
        let is_space = c == 32 || c == 10 || c == 13 || c == 9;
        if is_space {
            in_word = false;
        } else if !in_word {
            cnt += 1;
            in_word = true;
        }
    }

    std::io::stdout().write_all(&cnt.to_le_bytes()).unwrap();
}

File: go/word_count.go
package main

import (
	"encoding/binary"
	"io"
	"os"
)

func isSpace(b byte) bool {
	return b == 32 || b == 10 || b == 13 || b == 9
}

func main() {
	input, err := io.ReadAll(os.Stdin)
	if err != nil {
		os.Exit(1)
	}

	var cnt uint32
	inWord := false
	for _, ch := range input {
		if isSpace(ch) {
			inWord = false
			continue
		}
		if !inWord {
			cnt++
			inWord = true
		}
	}

	var out [4]byte
	binary.LittleEndian.PutUint32(out[:], cnt)
	if _, err := os.Stdout.Write(out[:]); err != nil {
		os.Exit(1)
	}
}
